src/schedule/zephyr_domain.c

```diff
@@ -168,6 +189,18 @@ static int zephyr_domain_register(struct ll_schedule_domain *domain,
 		k_timer_user_data_set(&zephyr_domain->timer, zephyr_domain);
 
 		k_timer_start(&zephyr_domain->timer, start, K_USEC(LL_TIMER_PERIOD_US));
+
+#if CONFIG_WDT_DW
+		if (watchdog[core]) {
+			/* Watchdog timeout configuration. */
+			const struct wdt_timeout_cfg watchdog_config = {
+				.window.max = 2 * LL_TIMER_PERIOD_US / 1000,
+			};
+
+			wdt_install_timeout(watchdog[core], &watchdog_config);
```

Comment:
I don't see the wdt ever uninstalled? Wouldn't the watchdog fire when the ll-scheduler ends? In most cases, the system will go to power-off state, but I'd still expect to have symmetry here that if watchdog is installed for a thread, it's removed when the thread exits cleanly.